### DISCUSSION TITLE
Battery related values all zero

### DISCUSSION THREAD

**Discussion by pickup7** [SBFspot/SBFspot#628] [Q&A]
Dear all, 
I am running SBFspot as a stable application since May 2022. Just upgraded to the latest version. What I never could achieve until now was to extract and MQTT transfer  the values for the Battery (BYD model 7,7 V) related values:
"**BatVol**": 0.000,"**BatAmp**": 0.000,"BatChaStt": 0.000
They unfortunalety remain zero although reasonable values are present and show up when I login to the Web based GUI of the SMA inverter. There it currently just reads:
- Aktueller Batterieladezustand  93 % 
- Batterietemperatur 17,0 °C 
- Batteriespannung 327,80 V  
- Batteriestrom  0,000 A  
- Betriebsstatus der Batterie Batterie Standby
A complete MQTT JSON-formatted message reads:
sbfspot {"Timestamp": "2023-03-30 14:34:02","SunRise": "2023-03-30 07:12:00","SunSet": "2023-03-30 19:58:00","InvTime": "2023-03-30 14:34:05","InvStatus": "Ok","InvTemperature": 0.000,"EToday": 5.862,"PACTot": 0.000,"MeteringWOut": 0.000,"MeteringWIn": 150.000,"MeteringWTot": 150.000,"BatVol": 0.000,"BatAmp": 0.000,"BatChaStt": 0.000}

Since at least the BaChaStt is of rather great importance to plan a my multi-level water heating in my scheduling software (depending on the Byttery Charge Status in percent) I would very much appreciate any hint or suggestion on why the values are not extracted / MQTT-forwarded to the data sink.

Thank you very much indeed.
Ulrich

**Comment by sbf-** [comment]
Can you post a debug (-d5 -v5) log pls?

**Comment by pickup7** [comment]
THX. Pls find log lines  debug/verbose 5 attached Rgds. Ulrich
[log.txt](https://github.com/SBFspot/SBFspot/files/11112740/log.txt)

**Comment by sbf-** [comment]
```
Device Class:     UNKNOWN CLASS
Device Type:      SUNNY TRIPOWER 8.0 SE
```
I see, it's a hybrid inverter. I have to do some fixes in order to make it work...

**Comment by silliewimons** [comment]
this is a message to 'pickup7'  - ulrich
- dear ulrich    thank you for your testing the  v397 version,
- while using v397  i'm confronted with the fact that on my system the csv 'spot-table' (with real-time info) is NOT CREATED, 
   after comparing my system with the log-file that you did post  i found that your configuration is identical with mine,
- difference you extract information directly to MQTT,  while I use  also  the csv-output (real-time-table)

??can you inform whether on your system the (real-time) spot-table is created   (spot is not the daily-table)
   on my system iwant to use the 'real-tile' output from sbfspot  in combination with the XXL-table (see below)

for your information: by using the webUI of inverter, i extract the 5minute table (nick-named XXL-table) 
   i found that one needs to ACTIVATE this option, by requesting once the 5minute 'data-output' via the webUI
   writing records into the XXL-table does only start after a first (output) request
   and there is a sily restriction, "to-day" output can not be retrieved - you can only consult information for past-days

thank you beforehand    sillie

**Comment by pickup7** [comment]
Sillie, 
I see a csv file which contains such

```
2023-04-01 09:52:02;SN: 3012373365;SUNNY TRIPOWER 8.0 SE;3012373365;714.000;707.000;704.000;3.000;3.000;3.000;235.500;234.300;233.300;2120.000;1.854;5683.339;50.040;0.000;0.000;Ok;13.000;15.000;307.500;5.544;0.000;4.000
2023-04-01 09:53:02;SN: 3012373365;SUNNY TRIPOWER 8.0 SE;3012373365;709.000;709.000;709.000;3.000;3.000;3.000;234.700;234.600;234.500;2130.000;1.890;5683.375;50.030;0.000;0.000;Ok;13.000;15.000;306.600;5.622;0.000;3.000
```

lines. 
Is it the "spot" one you are talking about ? It is located in the "log" directory and the naming convention is:
_Plant-Name_-Battery-YYYYMMDD.csv  

Rgds
Ulrich

**Reply by silliewimons** [reply]
to ulrich,     that is indeed the kind of csv file that i expected to be available in my system,

so i need to control my system to find where it might go wrong  

thank you for the assistance,     kr  sillie
